### PR TITLE
ENH: Wires up core-metrics(-phylo) to use diversity-lib

### LINKED ISSUE #214: core-metrics-[phylogenetic] n-jobs is inconsistent with help text
[forum x-ref](https://forum.qiime2.org/t/parallel-in-core-metrics-phylogenetic/4927)

I suspect something may have changed when we converted these to pipelines. From the type it looks like the range is 0-inf whereas the range for beta-phylogenetic is 1-inf, so we need to check the integration here as there's probably issues.

### PR DESCRIPTION
**Pull request by ChrisKeefe**
Closes #214

### CODE REVIEW

**Comment by ChrisKeefe** [comment]
image

**Comment by ChrisKeefe** [comment]
Failing pending addition of `q2-diversity-lib` to build env. https://github.com/qiime2/busywork/pull/101

**Comment by ChrisKeefe** [comment]
Tests failing due to absence of beta diversity measures. See https://github.com/qiime2/q2-diversity-lib/pull/6

File: q2_diversity/_core_metrics.py [hunk 1]
```diff
@@ -43,24 +46,24 @@ def core_metrics(ctx, table, sampling_depth, metadata, with_replacement=False,
...
 
     dms = []
-    dms += beta_phylogenetic(table=cr.rarefied_table, phylogeny=phylogeny,
-                             metric='unweighted_unifrac', n_jobs=n_jobs)
-    dms += beta_phylogenetic(table=cr.rarefied_table, phylogeny=phylogeny,
-                             metric='weighted_unifrac')
+    dms += unweighted_unifrac(table=cr.rarefied_table, phylogeny=phylogeny,
+                              n_jobs=n_jobs)
+    dms += weighted_unifrac(table=cr.rarefied_table, phylogeny=phylogeny,
+                            n_jobs=n_jobs)
```
Comment (thermokarst) [inline q2_diversity/_core_metrics.py:66]: These need to be using the param named `threads`, not `n_jobs`, in order to match what you have done in https://github.com/qiime2/q2-diversity-lib/pull/6

File: q2_diversity/_core_metrics.py [hunk 2]
```diff
@@ -42,25 +45,25 @@ def core_metrics(ctx, table, sampling_depth, metadata, with_replacement=False,
 
 
 def core_metrics_phylogenetic(ctx, table, phylogeny, sampling_depth, metadata,
-                              n_jobs=1):
-    alpha_phylogenetic = ctx.get_action('diversity', 'alpha_phylogenetic')
-    beta_phylogenetic = ctx.get_action('diversity', 'beta_phylogenetic')
+                              n_jobs_or_threads=1):
+    faith_pd = ctx.get_action('diversity_lib', 'faith_pd')
```
[the remark this answers is not in the document]
Comment (gwarmstrong) [reply q2_diversity/_core_metrics.py:49]: @thermokarst thanks for the heads up! q2-diversity-lib looks awesome!

[the remark this answers is not in the document]
Comment (thermokarst) [reply q2_diversity/_core_metrics.py:49]: That's all @ChrisKeefe's handiwork!

File: q2_diversity/plugin_setup.py [hunk 3]
```diff
@@ -723,7 +725,7 @@
     },
     parameter_descriptions={
         'metrics': ('The metrics to be measured. By default computes '
-                    'observed_otus, shannon, and if phylogeny is '
+                    'observed_features, shannon, and if phylogeny is '
```
Comment (thermokarst) [inline q2_diversity/plugin_setup.py:728]: Please revert - I don't believe we have changed the name of this metric here, yet.

```suggestion
                    'observed_otus, shannon, and if phylogeny is '
```

Bigger picture - we will just have references to both `observed_otus` and `observed_features` in this plugin until we sort everything out.

Comment (ChrisKeefe) [reply q2_diversity/plugin_setup.py:728]: Sorry for the noise. That shouldn't have been changed.

File: q2_diversity/_alpha/_method.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def non_phylogenetic_metrics():
             'dominance', 'doubles', 'enspie', 'esty_ci', 'fisher_alpha',
             'goods_coverage', 'heip_e', 'kempton_taylor_q', 'margalef',
             'mcintosh_d', 'mcintosh_e', 'menhinick', 'michaelis_menten_fit',
-            'observed_otus', 'osd', 'pielou_e', 'robbins', 'shannon',
+            'observed_features', 'osd', 'pielou_e', 'robbins', 'shannon',
```
Comment (thermokarst) [inline q2_diversity/_alpha/_method.py:35]: Please revert this change:

```suggestion
            'observed_otus', 'osd', 'pielou_e', 'robbins', 'shannon',
```
